common/utils/src/main/scala/org/apache/spark/util/SparkFileUtils.scala

```diff
@@ -168,6 +168,25 @@ private[spark] trait SparkFileUtils extends Logging {
     }
     Files.copy(src.toPath(), dst.toPath(), StandardCopyOption.REPLACE_EXISTING)
   }
+
+  /** Return true if the content of the files are equal or they both don't exist */
+  def contentEquals(file1: File, file2: File): Boolean = {
+    if (file1 == null && file2 != null || file1 != null && file2 == null) {
```

Comment:
Should this condition be ` if ((file1 == null && file2 != null) || (file1 != null && file2 == null)) {`

Reply:
Java's logical AND (`&&`) has higher precedence than logical OR (`||`). So, it's okay here.